lib/zlib.js

```diff
@@ -156,6 +156,56 @@ function flushCallback(level, strategy, callback) {
   }
 }
 
+// 1. Returns true for finite numbers
+// 2. Returns false for undefined and NaN
+// 3. Throws ERR_INVALID_ARG_TYPE for non-numbers
+// 4. Throws ERR_OUT_OF_RANGE for infinite numbers
+function checkFiniteNumber(number, name) {
+  if (Number.isFinite(number)) {
+    return true;  // is a valid number
+  } else {
+    // undefined or NaN
+    if (number === undefined || Number.isNaN(number)) {
+      return false;
+    }
+
+    // Other non-numbers
+    if (typeof number !== 'number') {
+      const err = new errors.TypeError('ERR_INVALID_ARG_TYPE', name,
+                                       'number', number);
+      Error.captureStackTrace(err, checkFiniteNumber);
+      throw err;
+    }
+
+    // Infinite numbers
+    const err = new errors.RangeError('ERR_OUT_OF_RANGE', name,
+                                      'a finite number', number);
+    Error.captureStackTrace(err, checkFiniteNumber);
```

Comment:
Right now this does not hide the `checkRangesOrGetDefault` frame anymore. This is problematic right now because the function is called from different functions and we can not just put in `checkRangesOrGetDefault`. I would probably just inline the `checkFiniteNumber` into `checkRangesOrGetDefault` and just have a bit of code copied for the single entry that does not fit into the pattern.

And it would be possible to combine this by doing:

```js
let err;
if (typeif number !== 'number') {
  err = ...
} else {
  err = ...
}
Error.captureStackTrace(err, fn);
throw err;
```